core/cog-view.c

```diff
@@ -111,6 +120,16 @@ cog_view_class_init(CogViewClass *klass)
         g_param_spec_boolean("use-key-bindings", NULL, NULL, TRUE,
                              G_PARAM_READWRITE | G_PARAM_CONSTRUCT | G_PARAM_EXPLICIT_NOTIFY | G_PARAM_STATIC_STRINGS);
 
+    /**
+     * CogView:viewport: (default-value null)
+     *
+     * Viewport where the CogView belongs.
+     *
+     * Since: 0.20
+     */
+    s_properties[PROP_VIEWPORT] =
+        g_param_spec_object("viewport", NULL, NULL, G_TYPE_OBJECT, G_PARAM_READWRITE | G_PARAM_CONSTRUCT);
```

Comment:
I would make the property read-only, and update the value of `priv->viewport` only from `cog_viewport_add()` and `cog_viewport_remove()`.

If we allow to use `cog_view_set_viewport()`, then we need to also move the view from one viewport to another and keep them in sync with the propery, which I think is more complicated than we need.

Reply:
Sounds reasonable